src/ui/frames.c

```diff
@@ -1492,8 +1492,6 @@ meta_frames_button_press_event (GtkWidget      *widget,
           op = op > META_GRAB_OP_CLICKING_MAXIMIZE_HORIZONTAL ? META_GRAB_OP_CLICKING_MAXIMIZE : op;
           break;
         case META_FRAME_CONTROL_UNMAXIMIZE:
-          op = META_GRAB_OP_CLICKING_UNMAXIMIZE + event->button - 1;
-          op = op > META_GRAB_OP_CLICKING_UNMAXIMIZE_HORIZONTAL ? META_GRAB_OP_CLICKING_UNMAXIMIZE : op;
```

Comment:
Can you explain? Looking at `META_FRAME_CONTROL_MAXIMIZE` I would say that `op = META_GRAB_OP_CLICKING_UNMAXIMIZE;` is the line that should be removed... Might be good idea to check history of this code.

Reply:
op is overwritten at the end of the case

Reply:
Yes, but are you sure that is right thing? Perhaps you just found source for some bug...